### PR TITLE
added cryptoHashConfig to deidentifyTemplate

### PR DESCRIPTION
**Pull request by JayS-crest**
added `crypto_hash_config` to `primitive_transformation` in `info_type_transformation` for resource `google_data_loss_prevention_deidentify_template`

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dlp: added `crypto_hash_config` to `google_data_loss_prevention_deidentify_template`
```

### CODE REVIEW

**Comment by JayS-crest** [comment]
I've checked the `pre-build-validation` failure, and that is due to several files. Checking the logs, I've found `Deidentify.yaml` has one error of trailing spaces from already available code. I've removed that and I guess the `pre-build-validation` won't have issues due to this resource file.

File: mmv1/third_party/terraform/tests/resource_data_loss_prevention_deidentify_template_test.go [hunk 1]
```diff
@@ -1369,6 +1418,56 @@ resource "google_data_loss_prevention_deidentify_template" "config" {
...
+      transformations {
+        info_types {
+          name = "CRYPTO_HASH_UNWRAPPED_UPDATED_EXAMPLE"
+        }
+
+        primitive_transformation {
+          crypto_hash_config {
+            crypto_key {
+              unwrapped {
+                key = "VVdWVWFGZHRXbkUwZERkM0lYb2xRdz09"
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_data_loss_prevention_deidentify_template_test.go:1447]: Just want to double-check that this & the transformation below are intentionally not changing the values?

**Review by melinath — CHANGES_REQUESTED** [review]
LGTM overall - it looks like the lint errors in the DeidentifyTemplate.yaml file might be resolved by https://github.com/GoogleCloudPlatform/magic-modules/pull/8098 so you could rebase after that's merged to see if it helps.

**Comment by JayS-crest** [comment]
I've updated the tests and rebased as well so I guess the linting error will be resolved.

File: mmv1/third_party/terraform/tests/resource_data_loss_prevention_deidentify_template_test.go (hunk 1 above)
Comment (melinath) [reply mmv1/third_party/terraform/tests/resource_data_loss_prevention_deidentify_template_test.go]: I'm okay with merging this as-is, just wanted to make sure it's intentional. Note that this means that we're not specifically testing that this field being updated works, but we still have good coverage overall.

**Review by melinath — APPROVED** [review]
LGTM - failing tests are unrelated

**Pull request merged**